doc/devel/contributing.rst

```diff
@@ -0,0 +1,422 @@
+looks at the keywords, they just get passed through the API to the
+artist constructor which looks for suitably named methods and calls
+them with the value.
+
+As a general rule, the use of ``**kwargs`` should be reserved for
+pass-through keyword arguments, as in the example above.  If all the
+keyword args are to be used in the function, and not passed
+on, use the key/value keyword args in the function definition rather
+than the ``**kwargs`` idiom.
+
+In some cases, you may want to consume some keys in the local
+function, and let others pass through.  You can ``pop`` the ones to be
+used locally and pass on the rest.  For example, in
+:meth:`~matplotlib.axes.Axes.plot`, ``scalex`` and ``scaley`` are
+local arguments and the rest are passed on as
+:meth:`~matplotlib.lines.Line2D` keyword arguments::
+
+  # in axes.py
+  def plot(self, *args, **kwargs):
+      scalex = kwargs.pop('scalex', True)
+      scaley = kwargs.pop('scaley', True)
+      if not self._hold: self.cla()
+      lines = []
+      for line in self._get_lines(*args, **kwargs):
+          self.add_line(line)
+          lines.append(line)
+
+Note: there is a use case when ``kwargs`` are meant to be used locally
+in the function (not passed on), but you still need the ``**kwargs``
+idiom.  That is when you want to use ``*args`` to allow variable
+numbers of non-keyword args.  In this case, python will not allow you
+to use named keyword args after the ``*args`` usage, so you will be
+forced to use ``**kwargs``.  An example is
+:meth:`matplotlib.contour.ContourLabeler.clabel`::
+
+  # in contour.py
+  def clabel(self, *args, **kwargs):
+      fontsize = kwargs.get('fontsize', None)
+      inline = kwargs.get('inline', 1)
+      self.fmt = kwargs.get('fmt', '%1.3f')
+      colors = kwargs.get('colors', None)
+      if len(args) == 0:
+          levels = self.levels
+          indices = range(len(self.levels))
+      elif len(args) == 1:
+         ...etc...
+
+.. _custom_backend:
+
+Developing a new backend
+------------------------
+
+If you are working on a custom backend, the *backend* setting in
+:file:`matplotlibrc` (:ref:`customizing-matplotlib`) supports an
+external backend via the ``module`` directive.  if
+:file:`my_backend.py` is a matplotlib backend in your
+:envvar:`PYTHONPATH`, you can set use it on one of several ways
+
+* in matplotlibrc::
+
+    backend : module://my_backend
+
+
+* with the :envvar:`MPLBACKEND` environment variable::
+
+    > export MPLBACKEND="module://my_backend"
+    > python simple_plot.py
+
+* from the command shell with the `-d` flag::
+
+    > python simple_plot.py -dmodule://my_backend
+
+* with the use directive in your script::
+
+    import matplotlib
+    matplotlib.use('module://my_backend')
+
+.. _sample-data:
+
+Writing examples
+----------------
+
+We have hundreds of examples in subdirectories of
+:file:`matplotlib/examples`, and these are automatically generated
```

Comment:
"these" is ambiguous; it sounds like the _examples_ are automatically generated, but it's really their outputs that are automatically generated.